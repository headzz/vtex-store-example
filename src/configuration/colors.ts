const BLACK = '#000000'
const DARK_GREY = '#5B5B5B'
const GREEN_JUSSI = '#03FFA5'
const GREY_LVL1 = '#F2F2F2'
const PINK_JUSSI = '#EE0E7D'
const TRANSPARENT = 'transparent'
const WHITE = '#ffffff'

export default {
  BLACK,
  DARK_GREY,
  GREEN_JUSSI,
  GREY_LVL1,
  PINK_JUSSI,
  WHITE,
  TRANSPARENT
}
