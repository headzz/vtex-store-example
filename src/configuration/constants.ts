export const BLACK = '#000000'
export const WHITE = '#ffffff'
export const GREY_LVL1 = '#F2F2F2'
export const GREEN_JUSSI = '#03FFA5'
export const DARK_GREY = '#5B5B5B'
export const PINK_JUSSI = '#EE0E7D'
