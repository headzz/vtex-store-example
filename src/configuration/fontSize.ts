const SIZE_TEXT_SMALL = '1.2rem'
const SIZE_TEXT_NORMAL = '1.6rem'

const SIZE_TITLE_SMALL = '3.2rem'
const SIZE_TITLE_NORMAL = '6.4rem'

export default {
  SIZE_TEXT_SMALL,
  SIZE_TEXT_NORMAL,
  SIZE_TITLE_NORMAL,
  SIZE_TITLE_SMALL
}
