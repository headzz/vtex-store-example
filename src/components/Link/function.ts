import { colors } from 'configuration'

export const checkDecoration = (
  border: boolean,
  background: boolean
): string => {
  if (border || background) {
    return ''
  }
  return `text-decoration: underline;
  -webkit-text-decoration-color: ${colors.PINK_JUSSI};
  text-decoration-color: ${colors.PINK_JUSSI};`
}
