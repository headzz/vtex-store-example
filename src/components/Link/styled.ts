import { colors, fontSize } from 'configuration'
import styled from 'styled-components'
import { checkDecoration } from './function'

type Props = {
  background: boolean
  border: boolean
  margin: boolean
}
export const LinkStyled = styled('a')<Props>`
  background-color: ${({ background }) =>
    background ? colors.GREEN_JUSSI : colors.TRANSPARENT};
  border-radius: 6px;
  border: ${({ border }) => (border ? `1px solid ${colors.BLACK}` : 'none')};
  color: ${colors.BLACK};
  font-size: ${fontSize.SIZE_TEXT_NORMAL};
  margin-left: ${({ margin }) => (margin ? '3.2rem' : '0')};
  outline: 0;
  padding: 1.2rem 2.4rem;
  text-decoration: none;
  :hover {
    ${({ border, background }) => checkDecoration(border, background)}
  }
`
