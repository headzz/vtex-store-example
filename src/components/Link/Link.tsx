import { LinkStyled } from './styled'

type Props = {
  background: boolean
  border: boolean
  href: string
  margin: boolean
}

const defaultProps: Props = {
  background: false,
  border: false,
  href: '',
  margin: false
}

/**
 * @component
 * @name Link
 * @description Defines Link component.
 * @namespace components components/Link
 * @param  {element} children Defining the text from the Link.
 * @param  {boolean} background defines if background is transparent or not
 * @param  {boolean} border defines if link has border
 * @param  {string} href Responsible for the redirection of the page
 * @param  {boolean} margin Says if there is a margin left on the component
 *
 */
const Link: React.FC<Props> = ({
  background,
  border,
  children,
  href,
  margin
}) => (
  <LinkStyled
    background={background}
    border={border}
    href={href}
    margin={margin}
  >
    {children}
  </LinkStyled>
)

Link.defaultProps = defaultProps

export default Link
