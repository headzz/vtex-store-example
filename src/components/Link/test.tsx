import { render, screen } from '@testing-library/react'
import { colors } from 'configuration'

import Link from './Link'

describe('Link', () => {
  it('should render component with redirect link', () => {
    render(
      <Link href="localhost" margin={false} background={false} border={false}>
        Teste
      </Link>
    )

    expect(screen.getByRole('link', { name: /Teste/i })).toMatchSnapshot()
    expect(screen.getByRole('link', { name: /Teste/i })).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /Teste/i })).toHaveAttribute(
      'href',
      'localhost'
    )
  })

  it('should render component link with margin', () => {
    render(
      <Link href="localhost" margin background={false} border={false}>
        Teste
      </Link>
    )

    expect(screen.getByRole('link', { name: /Teste/i })).toMatchSnapshot()
    expect(screen.getByRole('link', { name: /Teste/i })).toHaveStyle(
      'margin-left: 3.2rem'
    )
  })

  it('should render component link background color', () => {
    render(
      <Link href="localhost" margin={false} background border={false}>
        Teste
      </Link>
    )

    expect(screen.getByRole('link', { name: /Teste/i })).toMatchSnapshot()
    expect(screen.getByRole('link', { name: /Teste/i })).toHaveStyle(
      `background-color: ${colors.GREEN_JUSSI}`
    )
  })

  it('should render component link with border', () => {
    render(
      <Link href="localhost" margin={false} background={false} border>
        Teste
      </Link>
    )

    expect(screen.getByRole('link', { name: /Teste/i })).toMatchSnapshot()
    expect(screen.getByRole('link', { name: /Teste/i })).toHaveStyle(
      `border: 1px solid ${colors.BLACK}`
    )
  })
})
