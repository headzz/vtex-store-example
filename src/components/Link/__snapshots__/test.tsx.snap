// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render component link background color 1`] = `
.c0 {
  background-color: #03FFA5;
  border-radius: 6px;
  border: none;
  color: #000000;
  font-size: 1.6rem;
  margin-left: 0;
  outline: 0;
  padding: 1.2rem 2.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  class="c0"
  href="localhost"
>
  Teste
</a>
`

exports[`Link should render component link with border 1`] = `
.c0 {
  background-color: transparent;
  border-radius: 6px;
  border: 1px solid #000000;
  color: #000000;
  font-size: 1.6rem;
  margin-left: 0;
  outline: 0;
  padding: 1.2rem 2.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  class="c0"
  href="localhost"
>
  Teste
</a>
`

exports[`Link should render component link with margin 1`] = `
.c0 {
  background-color: transparent;
  border-radius: 6px;
  border: none;
  color: #000000;
  font-size: 1.6rem;
  margin-left: 3.2rem;
  outline: 0;
  padding: 1.2rem 2.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: #EE0E7D;
  -webkit-text-decoration-color: #EE0E7D;
  text-decoration-color: #EE0E7D;
}

<a
  class="c0"
  href="localhost"
>
  Teste
</a>
`

exports[`Link should render component with redirect link 1`] = `
.c0 {
  background-color: transparent;
  border-radius: 6px;
  border: none;
  color: #000000;
  font-size: 1.6rem;
  margin-left: 0;
  outline: 0;
  padding: 1.2rem 2.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: #EE0E7D;
  -webkit-text-decoration-color: #EE0E7D;
  text-decoration-color: #EE0E7D;
}

<a
  class="c0"
  href="localhost"
>
  Teste
</a>
`
